### PR TITLE
[kube-prometheus-stack] add alertmanagerConfigMatcherStrategy for alertmanager

### PR DESCRIPTION
**Pull request by jojotong**
#### What this PR does / why we need it
Prometheus operator release feature `alertmanagerConfigMatcherStrategy` by https://github.com/prometheus-operator/prometheus-operator/pull/5084 at `v0.61.0`, but not contains in this helm chart.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by jojotong** [comment]
@andrewgkew @gianrubio @gkarthiks @scottrigby @Xtigyro

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 43.2.1
+version: 43.2.2
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:24]: new featue should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (jojotong) [reply charts/kube-prometheus-stack/Chart.yaml]: Updated.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 43.2.1
+version: 43.3.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:24]:
```suggestion
version: 44.2.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**